flake.nix

```diff
@@ -249,20 +239,30 @@
             };
           };
 
-          checks.pre-commit-check = pre-commit-check;
-
           devShells = let
-            profillingShell = p: {
+            mkDevShells = p: {
               # `nix develop .#profiling` (or `.#ghc928.profiling): a shell with profiling enabled
               profiling = (p.appendModule {modules = [{enableLibraryProfiling = true;}];}).shell;
+              # `nix develop .#pre-commit` (or `.#ghc928.pre-commit): a shell with pre-commit enabled
+              pre-commit = let
+                pre-commit-check = inputs.pre-commit-hooks.lib.${system}.run {
+                  src = ./.;
+                  hooks = {
+                    fourmolu.enable = true;
```

Comment:
I'm wondering about including other tools in this list, such as `cabal-fmt` and `nixpkgs-fmt`. However, I think that should be a follow-up PR.

Reply:
I was thinking about this as well and I agree that this should happen in a separate PR if we decide to add those.